src/mesh/curved_mesh.jl

```diff
@@ -135,6 +190,46 @@ function bilinear_mapping(x, y, mesh)
 end
 
 
+# In 3D
+# Trilinear mapping from the reference element to the domain described by the faces
+function trilinear_mapping(x, y, z, mesh)
+  @unpack faces = mesh
+
+  x1 = faces[1](-1, -1) # maped from (-1,-1,-1)
+  @assert x1 ≈ faces[3](-1, -1) ≈ faces[5](-1, -1) "faces[1](-1, -1), faces[3](-1, -1) and faces[5](-1, -1) need to match at (-1,-1,-1) corner"
+
+  x2 = faces[2](-1, -1) #  maped from (1,-1,-1)
+  @assert x2 ≈ faces[3]( 1, -1) ≈ faces[5]( 1, -1) "faces[2](-1, -1), faces[3]( 1, -1) and faces[5]( 1, -1) need to match at (1,-1,-1) corner"
+  
+  x3 = faces[1]( 1, -1) # maped from (-1, 1,-1)
+  @assert x3 ≈ faces[4](-1, -1) ≈ faces[5](-1,  1) "faces[1]( 1, -1), faces[4](-1, -1) and faces[5](-1,  1) need to match at (-1,1,-1) corner"
+  
+  x4 = faces[2]( 1, -1) # maped from  (1, 1,-1)
+  @assert x4 ≈ faces[4]( 1, -1) ≈ faces[5]( 1,  1) "faces[2]( 1, -1), faces[4]( 1, -1) and faces[5]( 1,  1) need to match at (1,1,-1) corner"
+  
+  x5 = faces[1](-1,  1) # maped from (-1,-1, 1)
+  @assert x5 ≈ faces[3](-1,  1) ≈ faces[6](-1, -1) "faces[1](-1,  1), faces[3](-1,  1) and faces[6](-1, -1) need to match at (-1,-1,1) corner"
+
+  x6 = faces[2](-1,  1) # maped from  (1,-1, 1)
+  @assert x6 ≈ faces[3]( 1,  1) ≈ faces[6]( 1, -1) "faces[2](-1,  1), faces[3]( 1,  1) and faces[6]( 1, -1) need to match at (1,-1,1) corner"
+  
+  x7 = faces[1]( 1,  1) # maped from (-1, 1, 1)
+  @assert x7 ≈ faces[4](-1,  1) ≈ faces[6](-1,  1) "faces[1]( 1,  1), faces[4](-1,  1) and faces[6](-1,  1) need to match at (-1,1,1) corner"
+  
+  x8 = faces[2]( 1,  1) # maped from (1, 1, 1)
+  @assert x8 ≈ faces[4]( 1,  1) ≈ faces[6]( 1,  1) "faces[2]( 1,  1), faces[4]( 1,  1) and faces[6]( 1,  1) need to match at (1,1,1) corner"
+
+  return 0.125 * (x1 * (1 - x) * (1 - y) * (1 - z) +
+                  x2 * (1 + x) * (1 - y) * (1 - z) +
+                  x3 * (1 - x) * (1 + y) * (1 - z) +
+                  x4 * (1 + x) * (1 + y) * (1 - z) +
+                  x5 * (1 - x) * (1 - y) * (1 + z) +
+                  x6 * (1 + x) * (1 - y) * (1 + z) +
+                  x7 * (1 - x) * (1 + y) * (1 + z) +
+                  x8 * (1 + x) * (1 + y) * (1 + z) )
```

Comment:
```suggestion
  return 1/8 * (x1 * (1 - x) * (1 - y) * (1 - z) +
                x2 * (1 + x) * (1 - y) * (1 - z) +
                x3 * (1 - x) * (1 + y) * (1 - z) +
                x4 * (1 + x) * (1 + y) * (1 - z) +
                x5 * (1 - x) * (1 - y) * (1 + z) +
                x6 * (1 + x) * (1 - y) * (1 + z) +
                x7 * (1 - x) * (1 + y) * (1 + z) +
                x8 * (1 + x) * (1 + y) * (1 + z) )
```
This is more general and does not force the return type to be `Float64` (please correct me, @ranocha, if this is still not optimal)

Reply:
Writing `1/8` or `0.125` does not change anything. If you want to make it generic, you should divide by `8` at the end. However, that will be a division, which is usually slower than a multiplication. Thus, if you really want to be efficient and generic, you would need to use something like `one_eighth = one(T) / 8`, where `T` is an appropriate general type, e.g. `typeof(x)`.

Reply:
OK, thanks for the feedback. In that case, you can just leave it as it is if you want (from my perspective)